stock_orderpoint_mto_as_mts/models/product_product.py

```diff
@@ -0,0 +1,131 @@
+        orderpoint_obj = self.env["stock.warehouse.orderpoint"]
+        for warehouse in warehouses:
+            orderpoint = orderpoint_obj.with_context(active_test=False).search(
+                [
+                    ("product_id", "=", self.id),
+                    (
+                        "location_id",
+                        "=",
+                        warehouse._get_locations_for_mto_orderpoints().id,
+                    ),
+                ],
+                limit=1,
+                order="id",
+            )
+            if orderpoint and not orderpoint.active:
+                orderpoint.write(
+                    {"active": True, "product_min_qty": 0.0, "product_max_qty": 0.0}
+                )
+            elif not orderpoint:
+                vals = self._prepare_missing_orderpoint_vals(warehouse)
+                orderpoint = orderpoint_obj.create(vals)
+            orderpoints |= orderpoint
+        return orderpoints
+
+    def _prepare_missing_orderpoint_vals(self, warehouse):
+        self.ensure_one()
+        return {
+            "warehouse_id": warehouse.id,
+            "product_id": self.id,
+            "company_id": warehouse.company_id.id,
+            "product_min_qty": 0,
+            "product_max_qty": 0,
+            "location_id": warehouse._get_locations_for_mto_orderpoints().id,
+            "product_uom": self.uom_id.id,
+        }
+
+    def _ensure_default_orderpoint_for_mto(self):
+        """Ensure that a default orderpoint is created for the MTO products.
+
+        that have no orderpoint yet.
+        """
+        for product in self._filter_mto_products():
+            wh = product._get_warehouse_missing_orderpoint_for_mto()
+            if wh:
+                product._create_default_orderpoint_for_mto(wh)
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        products = super().create(vals_list)
+        products.sudo()._ensure_default_orderpoint_for_mto()
+        return products
+
+    def write(self, vals):
+        original_mto_products = self._filter_mto_products()
+        res = super().write(vals)
+        self._update_mto_products(original_mto_products, vals)
+        return res
+
+    def _update_mto_products(self, original_mto_products, vals):
+        if "company_id" in vals:
+            # Change company, must archive orderpoints
+            self.sudo()._archive_orderpoints_on_mto_removal()
+            self.sudo()._ensure_default_orderpoint_for_mto()
+            return
+        elif "route_ids" not in vals:
+            # No changes about is_mto
+            self.sudo()._ensure_default_orderpoint_for_mto()
+            return
+        mto_products = self._filter_mto_products()
+        # Find all MTO products that have been changed to Not MTO
+        products_to_update = original_mto_products - mto_products
+        if products_to_update:
+            products_to_update._archive_orderpoints_on_mto_removal()
```

Comment:
Hello @jbaudoux , as I knew and tested, `is_mto` is the computed field, so it didn't appear in `vals` of `def write()`
So `not vals.get("is_mto", True)` is not worked in this case, that is the reason why I used `route_ids` instead of `is_mto`

Reply:
Yes, I know it, [It is defined on product.template](https://github.com/OCA/product-attribute/blob/3a646a09de5e0bc90f3a59328396b2a3da9d4771/product_route_mto/models/product_template.py#L11), but it is the computed field (and readonly = True in this case), if we want to change it, we must write on the dependency field (`route_ids`, `categ_id` and `is_mto of the route` in this case)

After that, `is_mto` on product.template will appear in vals of `def _write()`, not in `def write`